### PR TITLE
Workspace: Added first constraints and resolved circular references

### PR DESCRIPTION
**Pull request by ffriedl89**
This PR is the first step towards resolving #570.

A couple of things to note here: 
* We ran into naming issues between the `examples` library in the libs folder and the `examples` library in tools folder the linting rules shipped by NX. 
* I moved the form-field tests into the input.spec file due to a cyclic dependency
* I will move the components folder into the libs folder in a separate PR. 
* We split up the testing library into 3 libs to control where to import from more granularly.

### CODE REVIEW

File: libs/testing/browser/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# testing-browser
+
+This library was generated with [Nx](https://nx.dev).
```
Comment (tomheller) [inline libs/testing/browser/README.md]: Cool

File: tslint.json [hunk 2]
```diff
@@ -61,6 +62,28 @@
...
+          },
+          {
+            "sourceTag": "scope:examples",
+            "onlyDependOnLibsWithTags": ["scope:components"]
+          }
+        ]
+      }
+    ],
+
+    // Custom Rules
```
Comment (NinaKammerlander) [inline tslint.json]: Does that work? I thought comments are not allowed in json files.

Comment (lukasholzer) [reply tslint.json]: in tslint it is allowed but in my opinion, they should be avoided.

**Comment by lukasholzer** [comment]
please fix failing unit test of the event chart

File: components/filter-field/src/filter-field-tag/filter-field-tag.spec.ts [hunk 3]
```diff
@@ -38,11 +38,10 @@ import { Platform } from '@angular/cdk/platform';
...
   createComponent,
   MockNgZone,
-} from '@dynatrace/barista-components/testing';
-import {
-  mockObjectProperty,
   mockGetComputedStyle,
-} from '@dynatrace/barista-components/testing/mock';
+} from '@dynatrace/barista-components/testing/browser';
+
+import { mockObjectProperty } from '@dynatrace/barista-components/testing/node';
```
Comment (lukasholzer) [inline components/filter-field/src/filter-field-tag/filter-field-tag.spec.ts]: this is not ideal to import inside a component from testing/node...

Comment (ffriedl89) [reply components/filter-field/src/filter-field-tag/filter-field-tag.spec.ts]: Its the spec file ;)

**Comment by ffriedl89** [comment]
This PR is currently blocked by the upgrade to nx 9 
#588

**Pull request merged**